validations: add onLimitReached deprecation warning

nfriedly opened the pull request:
<!--
	Hi there! Thanks for contributing! Please fill in this template to help us
	review and merge the PR as quickly and easily as possible!
-->

## Related Issues

Relates to https://github.com/express-rate-limit/express-rate-limit/discussions/372

## What Does This PR Do?

Adds a deprecation warning for use of the `onLimitReached` config option.

### Added

- `onLimitReached` deprecation warning


## Checklist

- [x] The issues that this PR fixes/closes have been mentioned above.
- [x] What this PR adds/changes/removes has been explained.
- [x] All library tests (`npm run test`) pass.
- [x] All added/modified code has been commented, and
      methods/classes/constants/types have been annotated with TSDoc comments.
- [x] If a new feature has been added or a bug has been fixed, tests have been
      added for the same.
- [x] Add new warning to error codes page in wiki
- [x] Add redirect for new warning

nfriedly committed:
validations: add onLimitReached deprecation warning

gamemaker1 commented on test/library/validation-test.ts:
```diff
@@ -167,6 +167,16 @@ describe('validations tests', () => {
 
+	describe('onLimitReached', () => {
+		it('should log a warning if onLimitReached is set', () => {
+			validations.onLimitReached(undefined)
+			expect(console.warn).not.toBeCalled()
+			validations.onLimitReached(() => {})
+			expect(console.warn).toBeCalled()
+			expect(console.error).not.toBeCalled()
```
Could you please split this into two tests?

gamemaker1 commented on readme.md:
```diff
@@ -430,8 +430,8 @@ const limiter = rateLimit({
 A (sync/async) function that accepts the Express `request` and `response`
-objects that is called when a client has reached their rate limit, and will be
-rate limited on their next request.
+objects that is called the on the request where a client has just exceeded their
```
👍🏾

gamemaker1 reviewed (approved):
Looks good; just one nit on the tests.

nfriedly committed:
split onLimitReached unit test in two

nfriedly commented:
Ok, I split up the test,  added the error code to the wiki, added a redirect, and triggered the warning locally to verify that it worked and the link went to the right place.

cyrille1804-a11y reviewed (commented):
Carrouselmaker Claude manages agent
